ouroboros-consensus-test/PLAN-statistical-testing.md

```diff
@@ -0,0 +1,159 @@
+# 2020 Stastical Testing Plan
+
+## Origin of Stochasticity
+
+There are two fundamental sources of stochasticity in the ThreadNet tests.
+
+  - Praos inherently includes stochasticity. The leader schedule is determined
+    by a nonce. The test setup generators should effectively sample the initial
+    nonce from a uniform distribution.
+
+  - Chain propagation delay introduces stochasticity into any Ouroboros
+    protocol. If different candidate chains happen to arrive at a node at
+    essentially the same time, the node may have to select amongst
+    equally-preferable chains. The tie-breaker mechanism ought to be at least
+    partially stochastic. (It appears to be uniform from my limited empirical
+    analysis.)
```

Comment:
Mock Praos just uses the `BlockNo` of the tips. For TPraos we look at the VRF hashes (it's more complicated). Do you think mock Praos needs a similar tie breaker?

Reply:
I think that's a question for the researchers. I do not recall seeing any tie breaking in the Ouroboros papers, so it currently seems preferable for the mock implementation to follow the papers as much as possible.